pandas/io/excel.py

```diff
@@ -465,7 +465,7 @@ def _parse_cell(cell_contents, cell_typ):
                 if not squeeze or isinstance(output[asheetname], DataFrame):
                     output[asheetname].columns = output[
                         asheetname].columns.set_names(header_names)
-            except StopIteration:
+            except EmptyDataError:
```

Comment:
The point is you need to put a halt in the current code and see what errors message that are being caught at the ``StopIteration``, and make sure that we are testing for each one. So when its changed you can be sure that you are catching the correct ones.